packages/ra-auth-cognito/src/ErrorMfaTotpRequired.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Error thrown when a user is required to enter a TOTP code.
+ */
+export class ErrorMfaTotpRequired extends Error {
+    constructor(message?: string) {
+        super(message);
+        Object.setPrototypeOf(this, ErrorMfaTotpRequired.prototype);
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** I don't understand the point of this line. Isn't that redundant with the `extends` behavior? Also, the [MDN web doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf) warns about implementing this.‏

Reply:
If I understand correctly, this code is indeed redundant thanks to TypeScript.
But since I'm not 100% sure, and it currently does not raise any error nor warning, I recommend we keep it.
This also has the benefit of being more consistent with the other similar implementations in RA.